src/validate-binding-behavior-base.ts

```diff
@@ -31,24 +34,46 @@ export abstract class ValidateBindingBehaviorBase {
     controller.registerBinding(binding, target, rules);
     binding.validationController = controller;
     const trigger = this.getValidateTrigger(controller);
-    // tslint:disable-next-line:no-bitwise
-    if (trigger & validateTrigger.change) {
+    const event =
+      (trigger & validateTrigger.blur) === validateTrigger.blur ? 'blur'
+        : (trigger & validateTrigger.focusout) === validateTrigger.focusout ? 'focusout'
+          : null;
+    const hasChangeTrigger = (trigger & validateTrigger.change) === validateTrigger.change;
+    binding.isDirty = !hasChangeTrigger;
+    binding.validatedOnce = hasChangeTrigger && event === null;
+    if (hasChangeTrigger) {
       binding.vbbUpdateSource = binding.updateSource;
       // tslint:disable-next-line:only-arrow-functions
       // tslint:disable-next-line:space-before-function-paren
       binding.updateSource = function (value: any) {
         this.vbbUpdateSource(value);
-        this.validationController.validateBinding(this);
+        this.isDirty = true;

```

Comment:
I haven't seen this `isDirty` being set to `false` anywhere else. I can understand that from the overall code, but it seems this doesn't allow "behavior reset". Ex: after a successful submission

Reply:
You are right. I also thought about this. I am not sure how to detect a "successful submission" only from the BB. We might need to change the validation controller for that. For example, this flag can be re/set when the binding property is validated. But in that case, the BB logic is not correctly encapsulated. 

My assumption is that usually a successful submission will be followed by a redirection. From that perspective, it might not be that big of an annoyance.

Reply:
Moreover, this is a property coming specifically from this BB. If we are encouraging community submission for new BBs, then we might need better API in place to support such cases.

Reply:
Thanks for sharing the thought. I agree that reseting could be difficult to implement in generic way. This looks good as it is